src/API/Router.ts

```diff
@@ -106,11 +109,13 @@ export class Router {
 
     this.logger.trace({ body }, 'POST /works')
 
-    if (!isClaim(body)) throw new IllegalArgumentException('Request Body must be a Claim.')
+    if (!isSignedVerifiableClaim(body)) throw new IllegalArgumentException('Request Body must be a Claim.')
 
-    if (!isWork(body)) throw new IllegalArgumentException(`Claim's type must be ${ClaimType.Work}, not ${body.type}`)
+    if (!isWork(body))
+      throw new IllegalArgumentException(`Claim's type must be ${ClaimType.Work}, not ${Object(body).type}`)
 
-    if (!isValidSignature(body)) throw new IllegalArgumentException("Claim's signature is incorrect.")
+    if (!this.verifiableClaimSigner.isValidSignature(body))
```

Comment:
Shouldn't this be using `await`? 

```ts
if (await !this.verifiableClaimSigner.isValidSignature(body))
```

Reply:
well `!(await this.verifiableClaimSigner.isValidSignature(body))` but yes